### PR TITLE
Feature/zip epic

### PR DESCRIPTION
**Pull request by Climbatize**
# Description
Adds zipped Miniapps support

## Links
[MINI-1087](https://jira.rakuten-it.com/jira/browse/MINI-1087)

# Checklist
- [X] I have read the [contributing guidelines](../CONTRIBUTING.md).
- [X] I wrote/updated tests for new/changed code
- [X] I removed all sensitive data before every commit, including API endpoints and keys
- [X] I ran `fastlane ci` without errors

### CODE REVIEW

File: Tests/Unit/MiniAppDownloaderTests.swift [hunk 1]
```diff
@@ -247,7 +248,28 @@ class MiniAppDownloaderTests: QuickSpec {
                             break
                         }
                     }
-                    expect(version).toEventually(equal("Mac"), timeout: 20)
+                    expect(version).toEventually(equal(versionId), timeout: 20)
+                }
+            }
+        }
+        describe("mini app downloader") {
```
Comment (ts-leojoseph-r) [inline Tests/Unit/MiniAppDownloaderTests.swift:255]: Should we also cover the failure scenario while unzipping ?

File: MiniApp/Classes/MiniAppDownloader.swift [hunk 2]
```diff
@@ -145,4 +155,17 @@ extension MiniAppDownloader: MiniAppDownloaderProtocol {
...
+
+    internal func unzipFile(fromURL destinationPath: String, to filePath: URL) {
+        if let directory = urlToDirectoryMap[destinationPath]?.miniAppDirectoryPath, filePath.fileExtension() == "zip" {
+            do {
+                try FileManager.default.unzipItem(at: filePath, to: directory, skipCRC32: true)
+                MiniAppLogger.d("MiniApp unzip time: \(Date().timeIntervalSince(time))")
+                try FileManager.default.removeItem(at: filePath)
+            } catch let err {
+                MiniAppLogger.e("error unzipping archive", err)
+                urlToDirectoryMap[destinationPath]?.completionHanlder(.failure(err))
```
[the remark this answers is not in the document]
Comment (Climbatize) [reply MiniApp/Classes/MiniAppDownloader.swift:167]: that makes sense

File: MiniApp/Classes/Utilities/Constants.swift [hunk 3]
```diff
@@ -4,4 +4,5 @@ struct Constants {
     static let javascriptInterfaceName = "MiniAppiOS"
     static let javascriptSuccessCallback = "MiniAppBridge.execSuccessCallback"
     static let javascriptErrorCallback = "MiniAppBridge.execErrorCallback"
+    static let offlineErrorCodeList: [Int] = [NSURLErrorNotConnectedToInternet, NSURLErrorTimedOut]
```
[the remark this answers is not in the document]
Comment (Climbatize) [reply MiniApp/Classes/Utilities/Constants.swift:7]: 👍

File: Tests/Unit/MiniAppDownloaderTests.swift (hunk 1 above)
Comment (Climbatize) [reply Tests/Unit/MiniAppDownloaderTests.swift:255]: I totally missed this

**Pull request merged**